### PR TITLE
feat(auth): plumb auth_enabled flag, add a settings page

### LINKED ISSUE #4192: [auth] plumb through `ENABLE_AUTH` setting through frontend

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #4192 

Plumbs through the auth enabled flag and displays UI in the new settings page for user and api key management
Screenshot 2024-08-12 at 12 59 58 PM

### CODE REVIEW

File: src/phoenix/server/main.py [hunk 1]
```diff
@@ -212,6 +218,8 @@ def _get_pid_file() -> Path:
         reference_inferences,
     )
 
+    authentication_enabled, auth_secret = get_auth_settings()
```
Comment (axiomofjoy) [inline src/phoenix/server/main.py:221]:
```suggestion
    authentication_enabled, _ = get_auth_settings()
```

File: src/phoenix/server/main.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 import phoenix.trace.v1 as pb
 from phoenix.config import (
     EXPORT_DIR,
+    get_auth_settings,
```
Comment (axiomofjoy) [inline src/phoenix/server/main.py:16]: Can also just import the `ENABLE_AUTH` constant.

**Review by axiomofjoy — APPROVED** [review]

File: app/src/pages/settings/SettingsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                description="The version of the Phoenix server"
+              />
+              <CopyToClipboardButtonWithPadding text={VERSION} />
+            </Flex>
+            <Flex direction="row" gap="size-100" alignItems="end">
+              <TextField
+                label="Python Version"
+                isReadOnly
+                value={`pip install 'arize-phoenix==${VERSION}'`}
+                description="The version of the Python client library to use to connect to this Phoenix"
```
Comment (axiomofjoy) [inline app/src/pages/settings/SettingsPage.tsx:57]:
```suggestion
                description="The version of the Python client library to use to connect to this instance of Phoenix"
```

File: src/phoenix/server/main.py (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/server/main.py:16]: I left it this way because you had some checks in there making sure both secret and enable is set?

**Pull request merged**